cura/API/Machines.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2019 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+from typing import Optional, Dict,  List, TYPE_CHECKING, Any
+from PyQt5.QtCore import QObject, pyqtSignal, pyqtSlot, pyqtProperty
+from UM.i18n import i18nCatalog
+from UM.Logger import Logger
+if TYPE_CHECKING:
+    from cura.CuraApplication import CuraApplication
+    from cura.PrinterOutput.PrinterOutputDevice import PrinterOutputDevice
+
+i18n_catalog = i18nCatalog("cura")
+
+##  The account API provides a version-proof bridge to use Ultimaker Accounts
+#
+#   Usage:
+#       ```
+#       from cura.API import CuraAPI
+#       api = CuraAPI()
+#       api.machines.addOutputDeviceToCurrentMachine()
+#       ```
+
+##  Since Cura doesn't have a machine class, we're going to make a fake one to make our lives a
+#   little bit easier.
+class Machine():
+    def __init__(self) -> None:
+        self.hostname = "" # type: str
+        self.group_id = "" # type: str
+        self.group_name = "" # type: str
+        self.um_network_key = "" # type: str
+        self.configuration = {} # type: Dict[str, Any]
+        self.connection_types = [] # type: List[int]
+
+class Machines(QObject):
+
+    def __init__(self, application: "CuraApplication", parent = None) -> None:
+        super().__init__(parent)
+        self._application = application
+
+    @pyqtSlot(result="QVariantMap")
+    def getCurrentMachine(self) -> Machine:
+        fake_machine = Machine() # type: Machine
+        global_stack = self._application.getGlobalContainerStack()
+        if global_stack:
+            metadata = global_stack.getMetaData()
+            if "group_id" in metadata:
+                fake_machine.group_id = global_stack.getMetaDataEntry("group_id")
+            if "group_name" in metadata:
+                fake_machine.group_name = global_stack.getMetaDataEntry("group_name")
+            if "um_network_key" in metadata:
+                fake_machine.um_network_key = global_stack.getMetaDataEntry("um_network_key")
```

Comment:
Proposal to change that? We're not going to interact directly with global stack.

Reply:
I think exposing the global stack is a very good idea to extend our API towards being able to change profiles, but you'd want a facade to direct API users towards the active global stack only.

Alternatively, we could expose the currently available connections through the API by just requesting it every time you make the API call, rather than having to update it every time it changes. Just remove the cache, really. It's not going to save a lot of processing time, because the network connection metadata shouldn't change often. Just once or twice per printer, probably.